packages/amplication-data-service-generator/src/plugin-manager.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  DsgPlugin,
+  Events,
+  EventsName,
+  PluginMap,
+} from "@amplication/code-gen-types";
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const emptyFunction = () => {};
+
+/**
+ * generator function that import the plugin requested by user
+ * @param pluginList
+ * @returns Plugin class
+ */
+async function* getPluginFuncGenerator(pluginList: DsgPlugin[]) {
+  try {
+    const pluginListLength = pluginList.length;
+    let index = 0;
+
+    do {
+      const packageName = pluginList[index].packageName;
+      const func = await import(packageName);
+
+      ++index;
+      if (!func.hasOwnProperty("default")) yield emptyFunction;
+
+      yield func.default;
+    } while (pluginListLength > index);
+  } catch (error) {
+    console.log(error); /// log error
+    return emptyFunction;
+  }
+}
+
+/**
+ * loop through all plugin list and set the plugin under each event
+ * @param pluginList
+ * @returns Array: Events[]
```

Comment:
I like the comment to explain what happens, but since we use TypeScript I believe we can remove `@param` and `@returns`, because people maintaining this will likely forget to update the JSDoc comment 😄